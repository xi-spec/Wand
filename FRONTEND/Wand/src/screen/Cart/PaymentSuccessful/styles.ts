import { StyleSheet } from 'react-native';
import { COLORS, SIZE, FONTS } from '../../../constants/theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 10,
    height: '100%',
    alignItems: 'center'
  },
  horizontal: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    padding: 10
  },
  image: {
    marginTop: '20%',
    width: 250,
    height: 250,
    marginBottom: '10%'
  },
  tittle: {
    fontFamily: FONTS.primaryMedium,
    fontSize: SIZE.big
  },
  textContainer: {
    marginTop: '10%'
  },
  text: {
    fontFamily: FONTS.extraLight,
    fontSize: SIZE.primary
  },
  thanks: {
    marginTop: '10%',
    fontFamily: FONTS.extraLight,
    fontSize: SIZE.primary
  },
  continueContainer: {
    width: '100%',
    height: 40,
    position: 'absolute',
    bottom: 25,
    backgroundColor: 'white'
  },
  continue: {
    width: '100%',
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: COLORS.primary,
    borderRadius: 5
  },
  continueText: {
    color: 'white',
    fontFamily: FONTS.logoRegular,
    fontSize: SIZE.primary

  }
});

export default styles;
