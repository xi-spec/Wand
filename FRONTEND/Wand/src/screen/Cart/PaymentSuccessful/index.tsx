import React from 'react';
import { Text, View, TouchableOpacity, Image, ActivityIndicator } from 'react-native';
import styles from './styles';
import { connect } from 'react-redux';
import { ShoppingCart, User } from '../../../models/initiaState';
import { IMAGES } from '../../../constants/images';

function PaymentSuccessful ({ navigation, cart, user }:
    {cart:ShoppingCart, user:User, navigation:{reset:Function}}) {
  return (
    cart.cartList.length !== 0
      ? <View style={[styles.container, styles.horizontal]}>
    <ActivityIndicator size="large"/>

  </View>
      : <View style={styles.container}>
    <Image style={styles.image} source={{ uri: IMAGES.paymentSuccessful }} />
    <Text style={styles.tittle}>Your Order has been accepted</Text>
    <View style={styles.textContainer}>
    <Text style={styles.text}>Your items has been placed and is on</Text>
    <Text style={styles.text}>itâ€™s way to being processed.</Text>
    </View>
     <Text style={styles.thanks}>Thanks!!</Text>

     <View style={styles.continueContainer}>
          <TouchableOpacity
          testID='continueShopping'
          disabled={!user.address || !user.card}
          onPress={() => navigation.reset({
            index: 0,
            routes: [{ name: 'Home' }]
          })}
           style={styles.continue}>
            <Text style={styles.continueText}>CONTINUE SHOPPING</Text>
          </TouchableOpacity>
        </View>

    </View>

  );
}

function mapStateToProps ({ cart, user }) {
  return {
    cart,
    user
  };
}

export default connect(mapStateToProps)(PaymentSuccessful);
