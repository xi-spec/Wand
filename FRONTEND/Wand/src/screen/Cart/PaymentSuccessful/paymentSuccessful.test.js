import { render, fireEvent, act } from '@testing-library/react-native';
import React from 'react';
import PaymentSuccessful from './index';
import ConfigureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

describe('Given a PaymentSuccessful component', () => {
  let navigation;
  const mockStore = ConfigureStore();
  const store = mockStore({
    cart: {
      cartList: [1, 2]
    },
    user: {
      address: {},
      card: {}
    }

  });
  beforeEach(() => {
    navigation = {
      navigate: jest.fn(),
      reset: jest.fn()
    };
  });

  describe('When is rendered', () => {
    test('Then is matched to snapshop', () => {
      const tree = render(<Provider store={store}> <PaymentSuccessful/></Provider>);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('When is rendered with empty cart', () => {
    test('Then is matched to snapshop', () => {
      const store = mockStore({
        cart: {
          cartList: []
        },
        user: {
          address: {},
          card: {}
        }

      });
      const tree = render(<Provider store={store}> <PaymentSuccessful/></Provider>);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('When CONTINUE SHOPPING button is pressed', () => {
    test('Then navigation.reset is invoked', () => {
      const store = mockStore({
        cart: {
          cartList: []
        },
        user: {
          address: {},
          card: {}
        }

      });
      const { getByTestId } = render(<Provider store={store}> <PaymentSuccessful navigation={navigation}/></Provider>);

      act(() => {
        fireEvent.press(getByTestId('continueShopping'));
      });

      expect(navigation.reset).toHaveBeenCalled();
    });
  });
});
