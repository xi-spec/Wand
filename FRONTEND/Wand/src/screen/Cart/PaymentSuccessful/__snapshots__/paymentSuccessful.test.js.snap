// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a PaymentSuccessful component When is rendered Then is matched to snapshop 1`] = `
Array [
  " ",
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "margin": 10,
        },
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "padding": 10,
        },
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
    />
  </View>,
]
`;

exports[`Given a PaymentSuccessful component When is rendered with empty cart Then is matched to snapshop 1`] = `
Array [
  " ",
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "margin": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://trello-attachments.s3.amazonaws.com/603be1b68ce9c654d30ae543/60465ae34c9420729026f19a/6e3717ef3a70d152203546808095dd3f/Captura_de_pantalla_2021-03-23_a_las_14.18.07.png",
        }
      }
      style={
        Object {
          "height": 250,
          "marginBottom": "10%",
          "marginTop": "20%",
          "width": 250,
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "MontserratMedium",
          "fontSize": 22,
        }
      }
    >
      Your Order has been accepted
    </Text>
    <View
      style={
        Object {
          "marginTop": "10%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "RalewayExtraLight",
            "fontSize": 18,
          }
        }
      >
        Your items has been placed and is on
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "RalewayExtraLight",
            "fontSize": 18,
          }
        }
      >
        itâ€™s way to being processed.
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontFamily": "RalewayExtraLight",
          "fontSize": 18,
          "marginTop": "10%",
        }
      }
    >
      Thanks!!
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "bottom": 25,
          "height": 40,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2E3192",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="continueShopping"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "RalewayRegular",
              "fontSize": 18,
            }
          }
        >
          CONTINUE SHOPPING
        </Text>
      </View>
    </View>
  </View>,
]
`;
